designer/server/src/common/templates/partials/navigation.njk

```diff
@@ -4,6 +4,7 @@
 {{ appServiceHeader({
   organisationName: "Defra",
   productName: "Forms Designer",
+  envTag: envTag,
```

Comment:
Shall we use Nunjucks for what it's good at?

By extending the [`config` context](https://github.com/DEFRA/forms-designer/blob/main/designer/server/src/common/nunjucks/context/index.js#L35-L38) we can do `config.env | titlecase` from inside a macro?

```njk
{%- macro envTag(env) -%}
  {%- switch env %}
    {% case "test" %}
      {% set classes = "govuk-tag--yellow" %}
    {% case "production" %}
      {% set classes = "govuk-tag--red" %}
    {% default %}
      {% set classes = "govuk-tag--grey" %}
  {% endswitch -%}

  {{ govukTag(
    text: upperFirst(env),
    classes: classes
  }) }}
{%- endmacro -%}
```

Could pass it in and add the macro alongside `littlePersonIcon()` already in the header:

```suggestion
  env: config.env,
```

Reply:
I'm easy but honestly I'm not convinced with the njk approach.

`njk title filter != upper case first letter` (looks like it [upper cases every word](https://github.com/mozilla/nunjucks/blob/master/nunjucks/src/filters.js#L508) like we saw a few months back)

Then there's the use of the undocumented jinja `switch` case statement (I wish they'd get round to this).

Reply:
Yeah assumed our environments were only single words

We've used `switch()` here already if it helps?

https://github.com/DEFRA/forms-designer/blob/f56d3af386f21db3056220ff0a2574fe8c35d4be/designer/server/src/common/components/heading/template.njk#L5-L15

Reply:
It makes me uncomfortable that it's not documented.

Also, I'm in the logic-less templates school of thought ala Handlebars so IMO, less NJK/template logic is better than more. JS will be around long after nunjucks has [been replaced](https://github.com/mozilla/nunjucks/issues/1433) or GDS decide to use something else.

Honestly though, I'm happy to go either way.

Reply:
I'm happy to swap it for if/else but for now the Nunjucks ship has sailed

For the next developer that joins, let's keep header logic inside the header